### PR TITLE
contrib/net/http Add WithQueryParams option to include query params in the value for http.url tag in trace

### PR DESCRIPTION
**Pull request by abhishp**
### What does this PR do?

* Add an option in the net/http tracer contrib package to toggle query parameters inclusion in the http.url span tag.

### Motivation

* The `httputil.TracerConfig` supports the `QueryParams` config but the same is not exposed as a `http.Option` function in the contrib/net/http package.

### Describe how to test/QA your changes

* Have added unit test which should test these changes.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

**Review by knusbaum — DISMISSED** [review]
LGTM

**Review by knusbaum — DISMISSED** [review]

**Comment by abhishp** [comment]
@knusbaum can you please approve the PR again? The review got dismissed after rebasing the PR.

File: contrib/net/http/http.go [hunk 1]
```diff
@@ -55,10 +55,11 @@ func (mux *ServeMux) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 	TraceAndServe(mux.ServeMux, w, r, &ServeConfig{
-		Service:  mux.cfg.serviceName,
-		Resource: resource,
-		SpanOpts: spanOpts,
-		Route:    route,
+		Service:     mux.cfg.serviceName,
+		Resource:    resource,
+		SpanOpts:    spanOpts,
+		Route:       route,
+		QueryParams: mux.cgf.queryParams,
```
Comment (dianashevchenko) [inline contrib/net/http/http.go:62]:
```suggestion
		QueryParams: mux.cfg.queryParams,
```

File: contrib/net/http/http_test.go [hunk 2]
```diff
@@ -398,6 +398,58 @@ func TestServerNamingSchema(t *testing.T) {
...
 
+func TestWithQueryParamsOption(t *testing.T) {
+	endpointURLWithQueryParams := "http://localhost/endpoint?key=value&otherKey=1"
+	tests := []struct{
+		queryParamsEnabled bool
+		expectedURLTag string
+	}{
+		{
+			queryParamsEnabled: true,
+			expectedURLTag: endpointURLWithQueryParams
```
Comment (dianashevchenko) [inline contrib/net/http/http_test.go:409]: Missing comma

File: contrib/net/http/http_test.go [hunk 3]
```diff
@@ -398,6 +398,58 @@ func TestServerNamingSchema(t *testing.T) {
...
+		},
+		{
+			queryParamsEnabled: false,
+			expectedURLTag: "http://localhost/endpoint"
+		}
+	}
+	
+
+	for _, test := range tests {
```
Comment (dianashevchenko) [inline contrib/net/http/http_test.go:418]: Url field doesn't exist in that struct

**Comment by dianashevchenko** [comment]
Hi @abhishp, there are some small issues with the code not compiling (missing commas, wrong struct field name), feel free to ping me for a quick review after you fix those 🙌

**Comment by darccio** [comment]
Closing in favour of an alternative implementation (by @mtoffl01).

**Pull request closed without merging**